src/global/tracking/TrackPropagation_factory.cc

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2023 Tyler Kutz
+
+#include "TrackPropagation_factory.h"
+
+#include <JANA/JEvent.h>
+#include <algorithms/tracking/JugTrack/TrackingResultTrajectory.hpp>
+#include <services/geometry/acts/ACTSGeo_service.h>
+
+#include <Acts/EventData/MultiTrajectoryHelpers.hpp>
+#include <Acts/Surfaces/RadialBounds.hpp>
+
+#include <extensions/spdlog/SpdlogExtensions.h>
+#include <services/log/Log_service.h>
+
+#include <edm4eic/TrackPoint.h>
+#include <edm4eic/TrackSegment.h>
+
+void eicrecon::TrackPropagation_factory::Init() {
+
+    auto app = GetApplication();
+
+    // SpdlogMixin logger initialization, sets m_log
+    InitLogger(app, GetTag());
+
+    auto acts_service = GetApplication()->GetService<ACTSGeo_service>();
+    m_track_propagation_algo.init(acts_service->actsGeoProvider(), logger());
+
+    m_geoSvc = app->template GetService<JDD4hep_service>();
+
+    SetPropagationSurfaces();
+
+}
+
+void eicrecon::TrackPropagation_factory::Process(const std::shared_ptr<const JEvent> &event) {
+
+    auto trajectories = event->Get<eicrecon::TrackingResultTrajectory>(GetInputTags()[0]);
```

Comment:
This is no the right type (anymore, ref #860) for the `CentralCKFTrajectories` passed to this factory, and it will fail compilation.

Either change this here and in the algorithm to use `edm4eic::Trajectory` (preferred) or change the collection to `CentralCKFActsTrajectories` (which will be remove in the future). The `edm4eic::Trajectory` collection is stored in the output ROOT file and can be used for associations, whereas the `eicrecon::TrackingResultTrajectory` collection does not get saved and cannot be used for associations.

Reply:
I agree that we should use edm4eic::Trajectory since it allows associations.  I'll change the algorithm and factory to use this.

Just to clarify though, it looks like TrackSegment is still associated with `edm4eic::Track`, not `edm4eic::Trajectory`.  So what would be the appropriate way to add the association to the trajectory?

Reply:
I don't think all pieces are in place yet to do the associations, so I'm not suggesting that the associations get added now. In particular, yes, we need to insert `edm4eic::Track` here, based on `edm4eic::Trajectory`, and some algorithms will only need to receive the tracks, not the trajectories.